### PR TITLE
[ODC-6378] dropdown to select pipelines

### PR DESCRIPTION
**Pull request by yozaam**
**Fixes**: 

https://issues.redhat.com/browse/ODC-6378

**Analysis / Root cause**: 

multiple `imagePipelines` are fetched, but there is no way to pick which one must be used

**Solution Description**: 

created a state for all the fetched `imagePipelines` and made a dropdown for all these (needed string format for keys in dropdown)

**Screen shots / Gifs for design review**: 

dropdownNames

**Unit test coverage report**: 

**Test setup:**

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/pipelines-plugin/src/components/import/pipeline/PipelineTemplate.tsx [hunk 1]
```diff
@@ -79,14 +80,31 @@ const PipelineTemplate: React.FC<PipelineTemplateProps> = ({ builderImages, exis
       const resourceSpecificPipeline = imagePipelines.find(
         (pl) => pl.metadata?.labels?.[labelType] === resources,
       );
+      setPipelineTemplates(imagePipelines);
```
[the remark this answers is not in the document]
Comment (yozaam) [reply frontend/packages/pipelines-plugin/src/components/import/pipeline/PipelineTemplate.tsx:83]: https://github.com/openshift/console/blob/8135af447bd8c3ee188af47abf10665b3bbb3498/frontend/packages/pipelines-plugin/src/components/import/pipeline/PipelineTemplate.tsx#L83

Currently(in master branch) we give preference to the pipeline with the matching resource type but if it is not found, we assign a pipleline having `labelType` property  

So should we only show the matching resource pipeline ? or include any pipeline with `.metadata.labels.[labelType]` present

**Comment by yozaam** [comment]
still unable to debug why we get a blank white dropdown when changing git url / resource type
dropdownHavingWhite

**Comment by karthikjeeyar** [comment]
> still unable to debug why we get a blank white dropdown when changing git url / resource type dropdownHavingWhite


Add a title prop (value: selectedTemplate) to the dropdown so it sets the value when the resource type is changed

File: frontend/packages/pipelines-plugin/src/components/import/pipeline/PipelineTemplate.tsx [hunk 2]
```diff
@@ -118,6 +171,14 @@ const PipelineTemplate: React.FC<PipelineTemplateProps> = ({ builderImages, exis
         name="pipeline.enabled"
         isDisabled={isPipelineAttached}
       />
+      {pipeline.enabled && (
+        <DropdownField
+          name="pipeline.templateSelected"
+          items={pipelineTemplateItems}
```
Comment (karthikjeeyar) [inline frontend/packages/pipelines-plugin/src/components/import/pipeline/PipelineTemplate.tsx:177]: This should solve the empty dropdown issue when resource type is altered. Refer my previous comment

```suggestion
          title={pipelineTemplateItems[pipeline.templateSelected]}
          items={pipelineTemplateItems}
```

Comment (yozaam) [reply frontend/packages/pipelines-plugin/src/components/import/pipeline/PipelineTemplate.tsx:177]: thank you it worked :D

File: frontend/packages/pipelines-plugin/src/components/import/pipeline/PipelineTemplate.tsx [hunk 3]
```diff
@@ -76,17 +77,48 @@ const PipelineTemplate: React.FC<PipelineTemplateProps> = ({ builderImages, exis
...
+          .map((pl) => pl.metadata?.name)
+          .sort()
+          .join(',') !==
+        candidatePipelines
+          .map((pl) => pl.metadata?.name)
+          .sort()
+          .join(',')
+      ) {
+        setPipelineTemplates(candidatePipelines);
+      }
```
Comment (karthikjeeyar) [inline frontend/packages/pipelines-plugin/src/components/import/pipeline/PipelineTemplate.tsx:99]:
```suggestion
    const getPipelineNames = (pipelines: PipelineKind[]): string =>
             pipelines
          .map((pl) => pl.metadata?.name)
          .sort()
          .join(',');  

      if (getPipelineNames(pipelineTemplates) !== getPipelineNames(candidatePipelines)) {
        setPipelineTemplates(candidatePipelines);
      }
      
```

Maybe move `getPipelineNames` helper function out of the useEffect ?

Comment (yozaam) [reply frontend/packages/pipelines-plugin/src/components/import/pipeline/PipelineTemplate.tsx]: thank you, updated

**Review by karthikjeeyar — APPROVED** [review]
Changes looks good and works as expected.
Thanks @yozaam

**Pull request merged**